src/Aspire.Hosting.Azure.ContainerApps/AzureContanierAppsInfrastructure.cs

```diff
@@ -0,0 +1,991 @@
+    }
+
+    /// <summary>
+    /// These are referencing outputs from azd's main.bicep file. We represent the global namespace in the manifest
+    /// by using {.outputs.property} expressions.
+    /// </summary>
+    private sealed class AzureContainerAppsEnvironment(string outputName) : IManifestExpressionProvider
+    {
+        public string ValueExpression => $"{{.outputs.{outputName}}}";
+
+        public static IManifestExpressionProvider MANAGED_IDENTITY_CLIENT_ID => GetExpression("MANAGED_IDENTITY_CLIENT_ID");
+        public static IManifestExpressionProvider MANAGED_IDENTITY_NAME => GetExpression("MANAGED_IDENTITY_NAME");
+        public static IManifestExpressionProvider MANAGED_IDENTITY_PRINCIPAL_ID => GetExpression("MANAGED_IDENTITY_PRINCIPAL_ID");
+        public static IManifestExpressionProvider AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID => GetExpression("AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID");
+        public static IManifestExpressionProvider AZURE_CONTAINER_REGISTRY_ENDPOINT => GetExpression("AZURE_CONTAINER_REGISTRY_ENDPOINT");
+        public static IManifestExpressionProvider AZURE_CONTAINER_APPS_ENVIRONMENT_ID => GetExpression("AZURE_CONTAINER_APPS_ENVIRONMENT_ID");
+        public static IManifestExpressionProvider AZURE_CONTAINER_APPS_ENVIRONMENT_DEFAULT_DOMAIN => GetExpression("AZURE_CONTAINER_APPS_ENVIRONMENT_DEFAULT_DOMAIN");
+
+        private static IManifestExpressionProvider GetExpression(string propertyExpression) =>
+            new AzureContainerAppsEnvironment(propertyExpression);
+    }
+
+    private sealed class ProjectResourceExpression(ProjectResource projectResource, string propertyExpression) : IManifestExpressionProvider
+    {
+        public string ValueExpression => $"{{{projectResource.Name}.{propertyExpression}}}";
+
+        public static IManifestExpressionProvider GetContainerImageExpression(ProjectResource p) =>
+            new ProjectResourceExpression(p, "containerImage");
+
+        public static IManifestExpressionProvider GetTargetPortExpression(ProjectResource p) =>
+            new ProjectResourceExpression(p, "containerPort");
+    }
+
+    /// <summary>
+    /// Generates expressions for the volume storage account. That azd creates.
+    /// </summary>
+    private sealed class VolumeStorageExpression(IResource resource, ContainerMountType type, string index) : IManifestExpressionProvider
+    {
+        public string ValueExpression => type switch
+        {
+            ContainerMountType.BindMount => $"{{{resource.Name}.bindMounts.{index}.storage}}",
+            ContainerMountType.Volume => $"{{{resource.Name}.volumes.{index}.storage}}",
+            _ => throw new NotSupportedException()
+        };
+
+        public static IManifestExpressionProvider GetVolumeStorage(IResource resource, ContainerMountType type, string index) =>
+            new VolumeStorageExpression(resource, type, index);
+    }
+
+    private sealed class PortAllocator(int startPort = 8000)
```

Comment:
Why do we have a default start port here, but we always use `10000` in the only place that creates this class?

Reply:
Yea, I should delete this.